src/MailPreviewMiddleware.php

```diff
@@ -16,18 +16,26 @@ class MailPreviewMiddleware
      */
     public function handle($request, Closure $next)
     {
-        $response = $next($request);
+        $addLinktoResponse = false;
 
         if (
             $request->hasSession() &&
-            $response instanceOf Response &&
             $previewPath = $request->session()->get('mail_preview_path')
         ) {
-            $this->addLinkToResponse($response, $previewPath);
+            $addLinktoResponse = true;
 
             $request->session()->forget('mail_preview_path');
```

Comment:
Maybe a theoretically question: Isn't this a change in the functionality? If $response isn't an instance of Response it will not add the link to the response but it will forget the session variable.

Reply:
You are right, I fixed this in the latest version.